src/OpenLoco/src/Objects/ObjectManager.cpp

```diff
@@ -559,6 +536,87 @@ namespace OpenLoco::ObjectManager
         return checksum == object.checksum;
     }
 
+    // 0x00471B95
+    void freeTemporaryObject()
+    {
+        if (_temporaryObject != nullptr && _temporaryObject != reinterpret_cast<Object*>(-1))
+        {
+            free(_temporaryObject);
+            _temporaryObject = reinterpret_cast<Object*>(-1);
+        }
```

Comment:
This is keeping -1 for vanilla compatibility, I presume? Perhaps add a note to that end?

Reply:
yeah unfortunately still needed.

Reply:
Could have
```cpp
static const auto kInvalidObject = reinterpret_cast<Object*>(-1);
```
instead.